packages/core/src/config/config.ts

```diff
@@ -26,6 +27,10 @@ export type ResolvedConfig = {
     chainId: number;
     /** RPC URL. Default: if available, a public RPC provider. */
     rpcUrl?: string;
+    /** Custom viem transport: https://viem.sh/docs/clients/transports/http.html
+     * If `undefined`, a default transport will be used based on `rpcUrl`.
+     */
+    transport?: Transport;
```

Comment:
If we go with a "one or the other" approach here, you can use [the `field?: never` trick](https://github.com/wagmi-dev/viem/blob/main/src/errors/base.ts#L3-L16) to produce a TS error if the user tries to provide both options.